### PR TITLE
Adding a CONTRIBUTING file

### PR DESCRIPTION
**Pull request by VLucet**
Following @florisvdh's suggestion, I am adding a small CONTRIBUTING file that explains the current plan for moving to support GRASS 8.0, as outlined by @rsbivand in #34.

### CODE REVIEW

**Review by rsbivand — APPROVED** [review]
Excellent, very clear, hope people read it!

**Review by veroandreo — APPROVED** [review]

**Review by florisvdh — COMMENTED** [review]
Nice @VLucet!

From:

> changes on `rgrass7` are merged into `main`

... my understanding is that `main` will not be worked on directly, but get its updates from `rgrass7` (as a repo maintenance job). So as far as **rgrass7** package is concerned, fixes should be contributed to the `rgrass7` branch. Am I right? Then that may not yet be so clear (IMHO) from:

- > The `main` branch remains the branch for current maintenance updates (bug fixes)
- > The `rgrass7` branch is the "target" maintenance source for the package `rgrass7`

Maybe switch both (reorder) and add more explicitly that `main` is not to be targeted by pull requests? If I don't get it right, then please clarify.

**Comment by VLucet** [comment]
@florisvdh Yes that is unclear, I've clarified. And yes, beyond this point, main wont be contributed to directly

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+### How to propose a change to rgrass
+
+To propose a change to `rgrass`, please consider the roles assigned to the different branches in this repository.
+
+* The `rgrass7` branch is the "target" maintenance source for the package `rgrass7` ("old" package). If you would like to contribute to `rgrass7`, please target your PRs and contributions to this branch.
+
+* The `rgrass` branch is the development branch for moving to support GRASS 8.0.
+
+* The `main` branch remains the branch for maintenance updates (bug fixes), until support for GRASS 8.0 is achieved (changes on `rgrass7` are merged into `main`). If you would like to contribute to `rgrass` ("new" package), please target your PRs and contributions to this branch.
```
Comment (florisvdh) [inline .github/CONTRIBUTING.md:13]: > If you would like to contribute to `rgrass` ("new" package), please target your PRs and contributions to this branch.

Perhaps move this sentence to the `rgrass` bullet (the one before `main`)? In the context of this bullet about `main`,  'to this branch' may seem like referring to `main`. Alternatively, you could replace 'to this branch' by 'to `rgrass`' and leave it here if you prefer that.

**Review by florisvdh — COMMENTED** [review]
Great! Just a single detail left :slightly_smiling_face:.

File: .github/CONTRIBUTING.md (hunk 1 above)
Comment (VLucet) [reply .github/CONTRIBUTING.md:13]: ah yes, I should move this info the bullet point about rgrass branch sorry!

**Review by florisvdh — APPROVED** [review]
:tada:

**Comment by florisvdh** [comment]
Probably best to also refer to `CONTRIBUTING.md` in `README.md`?

**Comment by VLucet** [comment]
Yes, done! I think the readme needs updating in general though.

**Comment by florisvdh** [comment]
> You’re all set—the `val-add-contributing` branch can be safely deleted. 

I've done this manually with the delete button.

Best automate branch deletion after PR-merge in the repo settings? @rsbivand

**Comment by florisvdh** [comment]
> automate branch deletion after PR-merge

We should be careful with this idea. It needs us to always merge `rgrass7` into `main` locally (`git merge`) and then push `main`, so never going via PRs for transferring commits between main and `rgrass7` / `rgrass`. But perhaps that makes most sense for our workflow, restricting PRs to the addition of new commits into either `rgrass7` or `rgrass`.

Any further ideas?

**Comment by rsbivand** [comment]
No, I don't think automatic deletion of branches is warranted unless they are simple issue -> PR -> merge & delete after review. I can (not knowing git  well) think of backporting settings before **rgrass7** is dropped completely where a branch merged with **rgrass** should also be merged with **rgrass7**, though I don't know if this is git-like. 

Yes, PR's to `rgrass` or `rgrass7` for onward merging to `main` makes sense. `main` should the be the reviewed and merged state of `rgrass7` until we drop further development of `rgrass7`.

**Pull request merged**